packages/table/src/common/internal/focusedCellUtils.ts

```diff
@@ -151,3 +263,16 @@ function getExpandedRegionIndices(
 
     return sourceIndex <= destinationIndex ? [sourceIndex, destinationIndex] : [destinationIndex, sourceIndex];
 }
+
+export function areFocusedRegionsEqual(left: FocusedRegion, right: FocusedRegion) {
+    if (left.type !== right.type) {
+        return false;
+    }
+
+    switch (left.type) {
+        case FocusMode.CELL:
+            return left.row === right.row && left.col === getFocusedColumn(right);
```

Comment:
**nit:** Can we use some type guards here to avoid the call to `getFocusedColumn`? e.g.

```ts
function isFocusedCell(region: FocusedRegion): region is FocusedCell {
    return region.type === FocusMode.CELL;
}

function isFocusedRow(region: FocusedRegion): region is FocusedRow {
    return region.type === FocusMode.ROW;
}

export function areFocusedRegionsEqual(left: FocusedRegion, right: FocusedRegion) {    
    if (isFocusedCell(left) && isFocusedCell(right)) {
        return left.row === right.row && left.col === right.col;
    } else if (isFocusedRow(left) && isFocusedRow(right)) {
        return left.row === right.row;
    } else {
        return false;
    }
}
```